andela/wger-space
# #166185762 Rest Api Registration

#### What does this PR do?
allows third-party applications to create users using the rest API . This is through making ```/api/v2/user/creation/```,```add-user-rest-api``` **and** ```list-user-rest-api```  functional.

#### Description of Task to be completed?
An application that is signed up as an ordinary user can create different users from its website using the wger custom commands that run in the terminal. These apps require a special key to be able to create other users.

#### How should this be manually tested?
- one should clone this repository and install the required dependencies
- log in to your local server supported front end and get a special key to allow you to create other users. It should look like this;

<img width="1434" alt="Screen Shot 2019-06-03 at 17 47 52" src="https://user-images.githubusercontent.com/43015966/58811059-c5aff380-8627-11e9-9d4f-e370fb223afc.png">

-  head on over to your terminal impersonating the third party application. start the local server in another terminal and populate the other terminal like:
``` python manage.py add-user-rest-api <special-key-from-website> <app-username>```
you should be able to see a green line of text stating 
```granted permission to create users```

At this point, one ought to run over to postman and test this endpoint for creating a user with the setup as demonstrated below on success:
<img width="1130" alt="Screen Shot 2019-06-06 at 14 31 20" src="https://user-images.githubusercontent.com/43015966/59030733-74dc0d00-886a-11e9-9aba-671075258004.png">


-  check a list of users created by that particular person or to check who created certain users, just keep server running and populate terminal with:
```<list-user-rest-api> <app-username> ``` and a fancy blue table will be output with two columns, one being for creator and username. Something like
<img width="776" alt="Screen Shot 2019-06-06 at 14 33 28" src="https://user-images.githubusercontent.com/43015966/59030003-63920100-8868-11e9-95e1-d95bebf91d0c.png">


#### Any background context you want to provide?
I got some of the ideas from past wger repo projects.
I also read [here](https://labofcoding.com/custom-django-management-commands/)
I also altered the Circle Ci badge to fix the current bug with this PR.
When merged, one will be able to use Heroku Terminal to run these commands

#### What are the relevant pivotal tracker stories?
[#166185762](https://www.pivotaltracker.com/story/show/166185762)
